lib/services/box_service.dart

```diff
@@ -46,21 +13,24 @@ class BoxService {
     required String boxType,
     bool isCanteen = false,
   }) async {
-    final query = _boxCollection
-        .where(Filter.or(Filter('charityId', isEqualTo: establishmentId),
-            Filter('canteenId', isEqualTo: establishmentId)))
-        .where('boxType', isEqualTo: boxType);
-
-    final querySnapshot = await query.get();
-
-    if (querySnapshot.docs.isEmpty) {
-      return false;
-    }
-
-    final box = querySnapshot.docs.single.data();
-
-    return isCanteen
-        ? box.quantityAtCanteen >= numberOfBoxes
-        : box.quantityAtCharity >= numberOfBoxes;
+    // TODO: Move to entity pair collection
```

Comment:
Again, can we please add a link to the JIRA ticket under which we will be addressing?

Reply:
Yes. Let's merge it.